crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -247,47 +319,114 @@ where
                     drop(epoch_sender_guard);
                 }
             }
+        }
+    }
 
-            let checkpoint_sender_guard = self.checkpoint_sender.lock().await;
+    async fn start_download_and_index_object_checkpoint(&self) -> Result<(), IndexerError> {
```

Comment:
sorry for getting to this PR late. But I'm not sure we want to keep duplicated code like this in main. Do we plan to clean this up asap? As time goes, it's going to be a lot harder to do. @gegaowp  @dmitri-perelman